src/MentionInputWidget.jsx

```diff
@@ -0,0 +1,321 @@
+
+        // We have to add some standard mendix classes to the rendered divs so they automatically look correct based on custom styles already existing
+        const mentionControl = this.nodeRef.current.querySelectorAll(".mentions__control");
+        mentionControl[0].classList.add("form-group");
+
+        document.addEventListener("mousedown", this.handleClickOutsideHandler, false);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener("mousedown", this.handleClickOutsideHandler, false);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps && prevProps === this.props) {
+            // Nothing changed
+        } else if (!this.props && !prevProps) {
+            // Props still empty
+        } else {
+            // check if widget is readonly
+            if (prevProps.valueAttribute.status === "loading" && this.props.valueAttribute.status === "available") {
+                this.checkReadOnly();
+            }
+            // valueAttribute changed
+            if (prevProps.valueAttribute.value !== this.props.valueAttribute.value) {
+                this.setState({ value: this.props.valueAttribute.value });
+                if (this.props.valueAttribute.value !== this.state.editedValue) {
+                    this.setState({ initialValue: this.props.valueAttribute.value });
+                }
+            }
+            // datasource is loaded so we can create the mentionslist
+            if (
+                this.state.readOnly === false &&
+                prevProps.datasource.status === "loading" &&
+                this.props.datasource.status === "available"
+            ) {
+                this.loadData();
+            }
+        }
+    }
+
+    checkReadOnly() {
+        const mentionHighlighter = this.nodeRef.current.querySelectorAll(".mentions__highlighter")[0];
+        const mentionInput = this.nodeRef.current.querySelectorAll(".mentions__input")[0];
+        if (this.props.valueAttribute.readOnly) {
+            this.setState({ readOnly: true });
+            mentionHighlighter.classList.add("form-control-static");
+            mentionInput.classList.add("form-control-static", "mx-textarea-input");
+        } else {
+            mentionHighlighter.classList.add("form-control");
+            mentionInput.classList.add("form-control", "mx-textarea-input");
+        }
+    }
+
+    loadData() {
+        // Function for loading the list of object into the mention suggestions.
+        console.debug("loadData started");
+        const suggestionlist = [];
+        this.props.datasource.items.forEach(mxObject => {
+            const objLabel = this.props.objLabel.get(mxObject).value;
+            const content = this.props.suggestionContent ? this.props.suggestionContent.get(mxObject) : objLabel;
+
+            const mentionObj = {
+                id: mxObject.id,
+                display: objLabel,
+                content: content
```

Comment:
```suggestion
                content
```
short hand notation preferred